### PR TITLE
feature(pu): add three variants of Bilinear classes and a FiLM class

### PR DESCRIPTION
**Pull request by puyuan1996**
## Description
- add three variants of Bilinear classes and a FiLM class into merge.py 
- add their unittest.

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/torch_utils/network/merge.py [hunk 1]
```diff
@@ -1,22 +1,176 @@
...
+        nn.init.uniform_(self.weight, -bound, bound)
+        nn.init.uniform_(self.bias, -bound, bound)
+
+    def forward(self, x, z):
+        # Using torch.einsum for the bilinear operation
+        out = torch.einsum('bi,oij,bj->bo', x, self.weight, z) + self.bias
+        return out.squeeze(-1)
+
+
+class TorchBilinear(nn.Bilinear):
```
Comment (PaParaZz1) [inline ding/torch_utils/network/merge.py:119]: why not try `TorchBilinear=nn.Bilinear`

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: ding/torch_utils/network/merge.py (hunk 1 above)
Comment (puyuan1996) [reply ding/torch_utils/network/merge.py:119]: Ok, this would indeed be simpler.

File: ding/torch_utils/network/merge.py [hunk 2]
```diff
@@ -1,22 +1,176 @@
...
+This file provides an implementation of several different neural network modules that are used for merging and
+transforming input data in various ways. The following components can be used when we are dealing with data from multiple modes,
 or when we need to merge multiple intermediate embedded representations in the forward process of a model.
 
-While SumMerge simply sums multiple data streams in the first dimension,
-VectorMerge provides three more complex weighted summations.
+The main classes defined in this code are:
+
+    - BilinearGeneral: This class implements a bilinear transformation layer that applies a bilinear transformation to incoming data,
+        as described in the "Multiplicative Interactions and Where to Find Them", published at ICLR 2020.
```
Comment (AltmanD) [inline ding/torch_utils/network/merge.py:9]: Can we put the link of  "Multiplicative Interactions and Where to Find Them" here?

Comment (puyuan1996) [reply ding/torch_utils/network/merge.py:9]: Ok

**Pull request merged**